rules-tests/TypeDeclaration/Rector/BooleanAnd/BinaryOpNullableToInstanceofRector/Fixture/binary_or.php.inc

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\BooleanAnd\BinaryOpNullableToInstanceofRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\BooleanAnd\BinaryOpNullableToInstanceofRector\Source\SomeInstance;
+
+function binaryOr(?SomeInstance $someClass)
+{
+    if ($someClass || $someClass->someMethod()) {
```

Comment:
For BinaryOr, the use case is on negation instead:

```php
if (! $someClass || $someClass->someMethod()) 
```

because truthy check always false on method call with or if not negated

Reply:
Also, that could be another case to cover. Could you add it?

Reply:
I will try, I am thinking that it must be negated, or that will cause phpstan error on the first try, I will check when back on pc, currently on mobile